woocommerce/woocommerce-ios
# [Orders with Coupons M6] Add Image thumbnail to collapsable product row

Closes: #10835 
#10844, #10845, and #10861 need to be merged first

## Description
This PR adds a product image to the collapsable product row component, rather than showing a `photo.stack.fill` placeholder.

For this, we extract the existing `KFImage` view that is currently used in the product rows into a separate component named  `ProductImageThumbnail`, which we reuse in both places.

## Testing instructions
This PR does not add any new logic or alters the proposed design of the collapsable component.
* With the `ordersWithCouponsM6` feature flag set to `false`:
  * Confirm that image thumbnails still appear when we add products to an Order.
* With the `ordersWithCouponsM6` set to `true`:
  * Confirm that image thumbnails now appear when we add products to an Order, rather than a placeholder.

## Screenshots
| Feature flag disabled | Feature flag enabled |
|--------|--------|
|  ![Simulator Screenshot - iPhone 15 - 2023-10-05 at 11 39 03](https://github.com/woocommerce/woocommerce-ios/assets/3812076/9ea7ee28-237c-40d5-b3e9-8778c93a217b)| ![Simulator Screenshot - iPhone 15 - 2023-10-05 at 11 38 11](https://github.com/woocommerce/woocommerce-ios/assets/3812076/602254c3-463f-4100-9415-5ace16f8c147)|
